components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/java/org/wso2/carbon/ndatasource/ui/NDataSourceHelper.java

```diff
@@ -498,22 +501,30 @@ private static void setDatasourceProperties(RDBMSDSXMLConfiguration rdbmsDSXMLCo
                         }
                 }
 	}
-	
-	public static String elementToString(Element element) {
-		try {
-			if (element == null) {
-				return null;
-			}
-		    Transformer transformer = TransformerFactory.newInstance().newTransformer();
-		    StringWriter buff = new StringWriter();
-		    transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-		    transformer.transform(new DOMSource(element), new StreamResult(buff));
-		    return buff.toString();
-		} catch (Exception e) {
-			log.error("Error while convering element to string: " + e.getMessage(), e);
-			return null;
-		}
-	}
+
+    public static String elementToString(Element element) {
+        try {
```

Comment:
Use try with resource
```suggestion
        try (StringWriter buff = new StringWriter()) {
```

Reply:
fixed